functions/api/report.js

```diff
@@ -26,6 +26,53 @@ router.get("/:branchId", async (request, response, next) => {
         }
     }
 
+    var { fromDate, toDate } = getFromAndToDate(request)
+    const branchReport = await getReportForBranch(branchId, fromDate, toDate)
+    response.status(200).json(branchReport)
+})
+
+/**
+ * @description Route to generate report for all products for all branches
+ * @returns Json object containing report for all products for all branches
+ */
+router.get("/all/branches", async (request, response, next) => {
+
+    const userRole = request.user.role
+    const userName = request.user.name
+
+    // Allow only admin/superadmin users to retrieve reports for all the branches
+    if (userRole === constants.BRANCH) {
```

Comment:
instead of this , why cant we do IsAdminOrSuperAdmin middleware check? Is there an issue with that?